### ISSUE TITLE
Refactor "manage content tab"

### ISSUE THREAD

**Issue by PatrickSkowronek** [h-da/geli#726]
# User Story

## Acceptance criteria:
- [ ] add button for create/add lecture
- [ ] add button for add unit
- [ ] add dialog for unit creation and editing
- [ ] the list view should not be changed to allow the drag and drop functionality

**Comment by grosspersky** [comment]
I think that using a dialog for this is a direct violation of the material style guide, since dialogs should only be used for interruptive messages but never for things like content creation.
https://material.io/guidelines/components/dialogs.html

**Comment by PatrickSkowronek** [comment]
@grosspersky yeah you right, but there is no other good solution for right now. @HPunktOchs and I had a good session yesterday. We have a better idea!

**Comment by grosspersky** [comment]
Maybe you should present your idea here in this issue. ;-)
Also you don't need dialogs for this, just open and display the edit forms the current way.
I think you just need more router outlets for this in the single components to display the matching content.

**Comment by grosspersky** [comment]
What is really important is that you still must be able to have a link to edit each lecture / unit to allow direct linking from the report pages to edit especially the units, which was a key requirement.

**Comment by PatrickSkowronek** [comment]
@grosspersky I making a sketch right now

**Comment by PatrickSkowronek** [comment]
@grosspersky 
![726-2](https://user-images.githubusercontent.com/32745673/40001010-88e2e184-578d-11e8-89e0-0e3fa77227c9.png)
![726-1](https://user-images.githubusercontent.com/32745673/40001011-88fb0480-578d-11e8-9fe4-1bfd11d65798.png)
This will improve the content tab

**Comment by grosspersky** [comment]
@PatrickSkowronek I'm still not able to see your plan by this two screens. Could you add some additional description on how the workflow will be implemented?

For example how which edit or add form will be accessed.

**Comment by PatrickSkowronek** [comment]
@grosspersky this is still a sketch, we did discuss in the meeting. We can talk about it tomorrow!

**Comment by dboschm** [comment]
This issue is blocked by the refactoring issue of unit form saving and general unit form structure #532

**Comment by PatrickSkowronek** [comment]
@grosspersky @d89 likes it what do you think?
![peek 2018-05-18 09-50](https://user-images.githubusercontent.com/32745673/40222808-0fa27ba8-5a81-11e8-897b-fa96e63d5ace.gif)

**Comment by grosspersky** [comment]
Overall this looks good, but it looks there are missing some features:
* Why does the import lecture button appear together with the _add lecture form_? An exported lecture would already contain a name and a description, so this is still a bit confusing. For lecture add / import I would still open a drop menu with the two entries _Add lecture_ and _Import lecture_.
* A import unit option is missing in the dropdown menu where you add your new unit

Is this still a sketch or did you actually commit this into a branch? If a branch exists I would take a deeper look in it locally.

**Comment by PatrickSkowronek** [comment]
I did push this https://github.com/h-da/geli/tree/feature/%23726-refactor-content-tab

> Why does the import lecture button appear together with the add lecture form? An exported lecture would already contain a name and a description, so this is still a bit confusing. For lecture add / import I would still open a drop menu with the two entries Add lecture and Import lecture.

Do you have a other idea. The mat-fab button should do one primary Action https://material.io/design/components/buttons-floating-action-button.html#types-of-transitions . Adding more would be a bad style. 

> A import unit option is missing in the dropdown menu where you add your new unit

In my opinion the right place is the action drop-down on the right. Where I did put it.
This is still in progress a few things are still not working or should not be displayed.
